contracts/GitcoinPassportDecoder.sol

```diff
@@ -0,0 +1,181 @@
+// SPDX-License-Identifier: GPL
+pragma solidity ^0.8.9;
+
+import { Initializable, OwnableUpgradeable } from "@openzeppelin/contracts-upgradeable/access/OwnableUpgradeable.sol";
+import { PausableUpgradeable } from "@openzeppelin/contracts-upgradeable/security/PausableUpgradeable.sol";
+import { UUPSUpgradeable } from "@openzeppelin/contracts/proxy/utils/UUPSUpgradeable.sol";
+import { Attestation, IEAS } from "@ethereum-attestation-service/eas-contracts/contracts/EAS.sol";
+
+import { GitcoinResolver } from "./GitcoinResolver.sol";
+
+import "hardhat/console.sol";
+
+/**
+ * @title GitcoinPassportDecoder
+ * @notice This contract is used to create the bit map of stamp providers onchain, which will allow us to score Passports fully onchain
+ */
+
+contract GitcoinPassportDecoder is 
+  Initializable,
+  UUPSUpgradeable,
+  OwnableUpgradeable,
+  PausableUpgradeable
+{
+  // The instance of the EAS contract.
+  IEAS eas;
+
+  // Mapping of version to provider arrays
+  mapping(uint32 => string[]) public providerVersions;
+
+  // Version number
+  uint32 public version;
+
+  // Address of the GitcoinResolver contract
+  address public gitcoinResolverAddress;
+
+  // Passport credential struct
+  struct Credential {
+    string provider;
+    bytes32 hash;
+    uint64 issuanceDate;
+    uint64 expirationDate;
+  }
+
+  function initialize(address _gitcoinResolverAddress) public initializer {
+    __Ownable_init();
+    __Pausable_init();
+
+    gitcoinResolverAddress = _gitcoinResolverAddress;
```

Comment:
Please use a setter for setting the GitcoinResolver, just as it is the case for EAS.